sdks/python/apache_beam/io/gcp/pubsub_test.py

```diff
@@ -82,25 +83,37 @@ def test_proto_conversion(self):
     self.assertEqual(m_converted.attributes, attributes)
 
   def test_eq(self):
-    a = PubsubMessage(b'abc', {1: 2, 3: 4})
-    b = PubsubMessage(b'abc', {1: 2, 3: 4})
-    c = PubsubMessage(b'abc', {1: 2})
+    publish_time_secs = 1520861821
+    publish_time_nanos = 234567000
+    publish_time = Timestamp(seconds=publish_time_secs,
+                             nanos=publish_time_nanos)
+    a = PubsubMessage(b'abc', {1: 2, 3: 4}, '1234567890', publish_time)
+    b = PubsubMessage(b'abc', {1: 2, 3: 4}, '1234567890', publish_time)
+    c = PubsubMessage(b'abc', {1: 2}, '1234567890', publish_time)
```

Comment:
Do we need to add additional tests cases to check new equality constraints on different message ids and timestamps?

Reply:
Yes, I will add to the hash and repr tests as well; a check to ensure that both a and b do not equal d/e with some differing message_id/publish_time data).